agent/command/git.go

```diff
@@ -548,21 +544,24 @@ func (c *gitFetchProject) executeLoop(ctx context.Context,
 
 	//Apply patches if this is a patch and we haven't already gotten the changes from a PR
 	if evergreen.IsPatchRequester(conf.Task.Requester) && conf.GithubPatchData.PRNumber == 0 {
-		for _, patchId := range c.AdditionalPatches {
-			p, err = comm.GetTaskPatch(ctx, td, patchId)
-			if err != nil {
-				return errors.Wrap(err, "unable to get additional patch")
-			}
-			if p == nil {
-				return errors.New("additional patch not found")
-			}
-			if err = c.getPatchContents(ctx, comm, logger, conf, p); err != nil {
-				return errors.Wrap(err, "Failed to get patch contents")
-			}
-			if err = c.applyPatch(ctx, logger, conf, reorderPatches(p.Patches)); err != nil {
-				return errors.Wrapf(err, "error applying patch '%s'", p.Id.Hex())
+		if !conf.Task.CommitQueueMerge {
```

Comment:
oops this was inverted. My test also conveniently had the wrong data to hide this, but I fixed that too

Reply:
For other tests, don't we also want the previous changes applied so that we know if the patch was really successful or not? (I might not be understanding what's happening here)

Reply:
The point of this block is to apply changes from patches before this patch in the commit queue. The additionalPatches variable won't be populated if this isn't a commit queue patch

Reply:
Oh will this be true for every task in a commit queue patch, not just the merge task?

Reply:
gah what I had initially was mostly right, I had forgotten why I added this here. So for normal tasks in a commit queue patch, we want to apply diffs from patches ahead of this one so that tests run with all the changes. Specifically for the merge, we don't want the diffs, because we only want to push the diff(s) that the author intended to merge. I fixed this, thank you for being suspicious of this